test.sh

```diff
@@ -21,5 +21,11 @@ yarn lint
 # Run unit tests
 yarn test
 
+# Ensure that Git repo is still clean
+if [[ -n "$(git status --porcelain)" ]]; then
+  echo "Git repository is not clean, make sure that all changes are committed"
```

Comment:
Generally speaking, if a git repo is clean before building the project, it should remain clean after the build as well. This was the intended purpose of the check above. Any artifacts generated during the build should be either committed or git-ignored.

Originally, I wanted to commit the generated API docs to the git repo, but then decided to git-ignore them for now, given the above mentioned follow-up that puts them to actual use.

> I know several folks run this script locally and this will be very annoying with uncommitted changes during development.

Stuff which is user specific, like TODO files for example, can be git-ignored, e.g. I have this in my `~/.gitignore`
```
**/vszocs-TODO*.txt
```
which allows me to put e.g. `vszocs-TODO-items.txt` file in any git repo I work on.

That said, I guess you're right that it may cause troubles for some folks, so I'll remove it.